pkg/server/container_execsync.go

```diff
@@ -72,6 +72,13 @@ type execOptions struct {
 
 // execInContainer executes a command inside the container synchronously, and
 // redirects stdio stream properly.
+// This function only returns when the exec process exits, this means that:
+// 1) As long as the exec process is running, the goroutine in the cri plugin
+// will be running and wait for the exit code;
+// 2) `kubectl exec -it` will hang until the exec process exits, even after io
```

Comment:
It would be nice to have -d detached option for kubectl exec.

Does it matter here "how" the io became detached wrt the way the container runtimes should behave, timeout.. interrupt... Maybe a todo is needed here.

Reply:
For example, if the `kubectl exec -it` process is deleted, the io will be detached, but the process will be left running.

Reply:
right that's one example... and based on that example what is the behavior, expected, documented, test results... yada yada. By here I meant here in the comment, did you have some known manual tests that you wanted to add to the comment based on how you got here to push the PR  :-)

Reply:
Done.